Doc/c-api/structures.rst

```diff
@@ -399,6 +399,37 @@ definition with the same method name.
    slot.  This is helpful because calls to PyCFunctions are optimized more
    than wrapper object calls.
 
+.. c:function:: PyObject * PyCFunction_New(PyMethodDef *ml, PyObject *self)
+
+   Turn *ml* into a Python callable object.  The *self* parameter will be
+   passed as *self* parameter to the C function in ``ml->ml_meth`` when
+   invoked, can be ``NULL``.
+
+   .. note::
+
+      This function "steals" a reference to *ml*.
+
+.. c:function:: PyObject * PyCFunction_NewEx(PyMethodDef *ml, PyObject *self, PyObject *module)
+
+   Same as :c:func:`PyCFunction_New`, but also allows setting the function
+   object's ``__module__`` attribute.  :attr:`!__module__` should be the name
+   of the module the function is defined in or ``None`` if unavailable.
+   See :attr:`function.__module__`.
+
+   .. note::
+
+      This function "steals" a reference to *ml*.
+
+.. c:function:: PyObject * PyCMethod_New(PyMethodDef *ml, PyObject *self, PyObject *module, PyTypeObject *cls)
+
+   Same as :c:func:`PyCFunction_NewEx`, but accept a **cls** parameter, which
+   will be passed as ``defining_class`` parameter to the C function.  Must be
+   set if :c:macro:`METH_METHOD` is set on ``ml->ml_flags``.
```

Comment:
```suggestion
   Same as :c:func:`PyCFunction_NewEx`, but accept a *cls* parameter, which
   will be passed as the *defining_class* argument to the C function.
   Must be set if :c:macro:`METH_METHOD` is set on ``ml->ml_flags``.
```

I think the last sentence reads a little bit weird; what must be set? The *cls* parameter?